### PR TITLE
Modify constructor to avoid calling it on anidated Schemy instances

### PR DESCRIPTION
**Pull request by JoaquinEduardoArreguez**
Modify constructor to avoid calling the constructor on anidated objects that already are Schemy instances, add jasmine test for the specific case

### CODE REVIEW

File: spec/basic.spec.js [hunk 1]
```diff
@@ -91,6 +91,43 @@ describe('Schemy basic validations', function() {
 		expect(result).toBe(true);
 	});
 
+	it("Should pass validation with correct child schemas using strict property", function () {
+    const nameSchema = new Schemy(
```
Comment (aeberdinelli) [inline spec/basic.spec.js:95]: Thank you for your PR! In order to merge it into the codebase, can you please fix indentation here? Schemy uses tab with 4 width. There is a .editorconfig to make this easier to solve in the root folder

**Review by aeberdinelli — CHANGES_REQUESTED** [review]
Thanks for your contribution! Please fix the requested changes and I will add the PR to next version of schemy!

**Comment by JoaquinEduardoArreguez** [comment]
thank you for your review, just pushed those indentation changes to the PR! :D

File: spec/basic.spec.js (hunk 1 above)
Comment (JoaquinEduardoArreguez) [reply spec/basic.spec.js]: Thanks for the review! do you know when will this new version be available??

File: package.json [hunk 2]
```diff
@@ -5,6 +5,7 @@
   "main": "index.js",
   "scripts": {
     "test": "jasmine-node --verbose --forceexit .",
+    "test-debugger": "node --inspect-brk node_modules/.bin/jasmine-node . --verbose --forceexit",
```
Comment (aeberdinelli) [inline package.json:8]: Nice addition!

**Review by aeberdinelli — APPROVED** [review]

**Comment by aeberdinelli** [comment]
Congratulations! I've published [version 3.0.6](https://www.npmjs.com/package/schemy/v/3.0.6) with your changes! Thank you very much. I've also added you as a contributor, you should appear soon in the "contributors" section of the npm package site. 🤓

**Comment by JoaquinEduardoArreguez** [comment]
> Congratulations! I've published [version 3.0.6](https://www.npmjs.com/package/schemy/v/3.0.6) with your changes! Thank you very much. [I've also added you as a contributor](https://github.com/aeberdinelli/schemy/commit/404e775332fa40986754834b1e0bc6d2e11266be#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R23), you should appear soon in the "contributors" section of the npm package site.

Thanks!! appreciate it! 😄

**Pull request merged**